.github/workflows/release.yml

```diff
@@ -5,53 +5,51 @@ on:
   - workflow_dispatch
 
 jobs:
-  create_release:
-    # https://github.com/actions/create-release/issues/14#issuecomment-555379810
-    runs-on: ubuntu-20.04
-    steps:
-    - name: Create Release
-      id: create_release
-      uses: actions/create-release@v1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        tag_name: ${{ github.ref }}
-        release_name: Draft Release
-        body: ''
-        draft: true
-    - name: Save Upload URL to File
-      run: printf "%s" "${{ steps.create_release.outputs.upload_url }}" > upload_url.txt
-    - name: Save Upload URL as Artifact
-      uses: actions/upload-artifact@v2
-      with:
-        name: upload_url
-        path: upload_url.txt
-
   build:
-    needs: create_release
     strategy:
       matrix:
-        os: [ubuntu20.04-aarch64, ubuntu20.04, ubuntu18.04]
+        os: [ubuntu22.04, ubuntu20.04, ubuntu18.04]
     runs-on: macos-12
     steps:
-    - uses: actions/checkout@v2
-    - name: Fetch Upload URL
-      uses: actions/download-artifact@v2
-      with:
-        name: upload_url
+    - uses: actions/checkout@main
     - name: Build Release
       id: build_release
       run: |
-        echo "::set-output name=upload_url::$(cat upload_url.txt)"
-        rm upload_url.txt
         ./create-toolchain ${{ matrix.os }}
-        echo "::set-output name=asset_name::"$(basename packages/*)
-    - name: Upload Release
-      uses: actions/upload-release-asset@v1
-      env:
+
+    - uses: actions/upload-artifact@main
+      with:
+        name: ${{ matrix.os }}-toolchain
+        path: packages/*-${{ matrix.os }}.tar.xz
+
+  upload:
+    needs: build
+    runs-on: ubuntu-latest
+    env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+    - uses: actions/checkout@main
+    - name: Download Artifacts
+      uses: actions/download-artifact@main
+
+    - name: Set TAG Variable
+      run: |
+        echo "TAG=${GITHUB_REF#refs/tags/}" >> "$GITHUB_ENV"
+
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
       with:
-        upload_url: ${{ steps.build_release.outputs.upload_url }}
-        asset_path: packages/${{ steps.build_release.outputs.asset_name }}
-        asset_name: ${{ steps.build_release.outputs.asset_name }}
-        asset_content_type: application/gzip
+        tag_name: ${{ env.TAG }}
+        release_name: Draft Release
+        body: ''
+        draft: true
+
+    - name: Attach Builds To Release
+      run: |
+        set -x
+        toolchains=()
+        for tc in **/*.tar.xz; do
+          toolchains+=("-a" "$tc")
+        done
+        hub release edit "${toolchains[@]}" -m "Draft Release" "$TAG"
```

Comment:
It's preferable to have each toolchain be added to the release as it builds, so that one build failing doesn't affect all of them. Using `hub` and `$GITHUB_ENV` are good changes, but if you could maintain the incremental behaviour that would be great.

Reply:
Implemented incremental behavior but switched to `gh` cli as `hub` appeared to have a couple of issues.